src/python/pants/option/subsystem.py

```diff
@@ -45,9 +75,42 @@ class Subsystem(metaclass=ABCMeta):
 
     _scope_name_re = re.compile(r"^(?:[a-z0-9_])+(?:-(?:[a-z0-9_])+)*$")
 
+    class EnvironmentAware(metaclass=ABCMeta):
```

Comment:
Adding a docstring for this class containing some of the description of this PR (essentially: how to implement this API) would be good.